Jenkinsfile

```
@@ -0,0 +1,141 @@
+    def deploymentName = "$imageName-$tag"
+    sh "helm delete --purge $deploymentName || echo error removing deployment $deploymentName"
+    sh "kubectl delete namespaces $deploymentName || echo error removing namespace $deploymentName"
+  }
+}
+
+def publishChart(imageName) {
+  // jenkins doesn't tidy up folder, remove old charts before running
+  sh "rm -rf helm-charts"
+  sshagent(credentials: ['helm-chart-creds']) {
+    sh "git clone git@gitlab.ffc.aws-int.defra.cloud:helm/helm-charts.git"
+    dir('helm-charts') {
+      sh 'helm init -c'
+      sh "helm package ../helm/$imageName"
+      sh 'helm repo index .'
+      sh 'git config --global user.email "buildserver@defra.gov.uk"'
+      sh 'git config --global user.name "buildserver"'
+      sh 'git checkout master'
+      sh 'git add -A'
+      sh "git commit -m 'update $imageName helm chart from build job'"
+      sh 'git push'
+    }
+  }
+}
+
+node {
+  checkout scm
+  stage('Set branch, PR, and containerTag variables') {
+    (branch, pr, containerTag, mergedPrNo) = getVariables(repoName)
+    if (pr ) {
+      sh "echo Building $pr"
+    } else if (branch == "master") {
+      sh "echo Building master branch"
+    } else {
+      currentBuild.result = 'ABORTED'
+      error('Build aborted - not a PR or a master branch')
+    }
+  }
+  stage('Build test image') {
+    buildTestImage(imageName, BUILD_NUMBER)
+  }
+  stage('Run tests') {
+    runTests(imageName, BUILD_NUMBER)
+  }
+  // note: there should be a `build production image` step here,
+  // but the docker file is currently not set up to create a production only image
+  stage('Push container image') {
+    pushContainerImage(registry, regCredsId, imageName, containerTag)
+  }
+  if (pr != '') {
+    stage('Helm install') {
+      withCredentials([
+          string(credentialsId: 'messageQueueHostPR', variable: 'messageQueueHost'),
+          usernamePassword(credentialsId: 'scheduleListenPR', usernameVariable: 'scheduleQueueUsername', passwordVariable: 'scheduleQueuePassword'),
+          usernamePassword(credentialsId: 'paymentListenPR', usernameVariable: 'paymentQueueUsername', passwordVariable: 'paymentQueuePassword'),
+          string(credentialsId: 'postgresExternalNamePaymentsPR', variable: 'postgresExternalName'),
+          usernamePassword(credentialsId: 'postgresPaymentsPR', usernameVariable: 'postgresUsername', passwordVariable: 'postgresPassword'),
+        ]) {
+        def extraCommands = "--values ./helm/ffc-demo-payment-service/jenkins-aws.yaml --set name=ffc-demo-$containerTag,container.messageQueueHost=\"$messageQueueHost\",container.scheduleQueueUser=\"$scheduleQueueUsername\",container.scheduleQueuePassword=\"$scheduleQueuePassword\",container.paymentQueueUser=\"$paymentQueueUsername\",container.paymentQueuePassword=\"$paymentQueuePassword\",postgresExternalName=\"$postgresExternalName\",postgresUsername=\"$postgresUsername\",postgresPassword=\"$postgresPassword\""
```

I think the name here needs changing to ffc-demo-payment-service
```suggestion
        def extraCommands = "--values ./helm/ffc-demo-payment-service/jenkins-aws.yaml --set name=ffc-demo-payment-service-$containerTag,container.messageQueueHost=\"$messageQueueHost\",container.scheduleQueueUser=\"$scheduleQueueUsername\",container.scheduleQueuePassword=\"$scheduleQueuePassword\",container.paymentQueueUser=\"$paymentQueueUsername\",container.paymentQueuePassword=\"$paymentQueuePassword\",postgresExternalName=\"$postgresExternalName\",postgresUsername=\"$postgresUsername\",postgresPassword=\"$postgresPassword\""
```